lib/kitchen/driver/ec2.rb

```diff
@@ -317,29 +316,40 @@ def tag_server(server)
       end
 
       def wait_until_ready(server, state)
-        wait_log = proc do |attempts|
-          c = attempts * config[:retryable_sleep]
-          t = config[:retryable_tries] * config[:retryable_sleep]
-          info "Waited #{c}/#{t}s for instance <#{state[:server_id]}> to become ready."
-        end
-        begin
-          server.wait_until(
-            :max_attempts => config[:retryable_tries],
-            :delay => config[:retryable_sleep],
-            :before_attempt => wait_log
-          ) do |s|
-            hostname = hostname(s, config[:interface])
-            # Euca instances often report ready before they have an IP
-            s.exists? && s.state.name == "running" && !hostname.nil? && hostname != "0.0.0.0"
-          end
-        rescue ::Aws::Waiters::Errors::WaiterFailed
-          error("Ran out of time waiting for the server with id [#{state[:server_id]}]" \
-            " to become ready, attempting to destroy it")
-          destroy(state)
-          raise
+        wait_with_destroy(server, state, "to become ready") do |aws_instance|
+          hostname = hostname(aws_instance, config[:interface])
+          # We aggressively store the hostname so if the process fails here
+          # we still have it, even if it will change later
+          state[:hostname] = hostname
```

Comment:
My fix is to store the hostname inside the method were we are updating the `server` (`aws_instance`) object